src/inausoft.netCLI/InvalidOptionException.cs

```diff
@@ -6,9 +6,12 @@ public class InvalidOptionException : Exception
     {
         public string OptionName { get; }
 
-        public InvalidOptionException(string optionName, string message = null) : base(message)
+        public string CommandName { get; }
+
+        public InvalidOptionException(string commandName, string optionName, string message = null) : base(message)
```

Comment:
I would consider moving the default message to exception constructor and introduce two ctors:

```cs
public InvalidOptionException(string commandName, string optionName)
: this(commandName, optionName, $"<message here>")
...

public InvalidOptionException(string commandName, string optionName, string message)
: base(message)
...
```

I think that in most cases you would be happy with default error message. Also if you don't provide any message, will you end up with an exception without a message?

Reply:
Agreed.